Tests/MongoSwiftSyncTests/CrudTests.swift

```diff
@@ -68,11 +68,22 @@ final class CrudTests: MongoSwiftTestCase {
     }
 
     func testCrudUnified() throws {
-        let files = try retrieveSpecTestFiles(specName: "crud", subdirectory: "unified", asType: UnifiedTestFile.self)
+        let files = try retrieveSpecTestFiles(
+            specName: "crud",
+            subdirectory: "unified",
+            excludeFiles: [
```

Comment:
when implementing the unified runner, I chose to just silently ignore any cases where update pipelines were used while decoding, setting them to an empty document (see changes in files below), and rely on the skip list below to not actually run the tests.

when syncing in a number of new tests using update pipelines as part of this work, this made it very challenging to debug the various tests that were failing, as update commands were being sent to the server but with an empty update document, which led to a variety of different failures in those tests.

I've now switched to erroring if an array pipeline is encountered, and just skipping all of the relevant files here so we don't bother trying to decode them. every file utilizing update pipelines is exclusively composed of tests using update pipelines, so we aren't losing any coverage by taking the "skip the whole file" approach.

relatedly, I'm wondering if we want to think about prioritizing SWIFT-560 since update pipelines have been coming up in spec tests a lot lately and we're the only driver to not yet support them, so I've moved it back into needs triage

Reply:
This approach to incompatible formats seems like a nice improvement. And sounds good about the update pipeline work.